torchaudio/transforms.py

```diff
@@ -276,14 +276,24 @@ def __init__(self,
         self.f_min = f_min
         self.norm = norm
         self.mel_scale = mel_scale
+        self.n_stft = n_stft
 
         assert f_min <= self.f_max, 'Require f_min: {} < f_max: {}'.format(f_min, self.f_max)
 
-        fb = torch.empty(0) if n_stft is None else F.create_fb_matrix(
+        fb = torch.empty(0) if self.n_stft is None else F.create_fb_matrix(
             n_stft, self.f_min, self.f_max, self.n_mels, self.sample_rate, self.norm,
             self.mel_scale)
         self.register_buffer('fb', fb)
 
+    def __prepare_scriptable__(self):
+        r"""
+        Prepare self to be scripted
+        Returns:
+            MelScale: self
+        """
+        assert self.n_stft is not None, ValueError("n_stft must be provided at construction")
```

Comment:
I realized that checking `self.fb.numel() != 0`, in the similar way as in the `forward` method, is more appropriate here. Can you change it? My original direction was not precise.